test/matcherexception.cpp

```diff
@@ -0,0 +1,30 @@
+#include "matcherexception.h"
+
+#include "3rd-party/catch.hpp"
+
+using namespace newsboat;
+
+extern "C" {
+	MatcherErrorFfi rs_get_test_attr_unavail_error();
+	MatcherErrorFfi rs_get_test_invalid_regex_error();
+}
+
+TEST_CASE("Can be constructed from Rust error returned over FFI",
+	"[MatcherException]")
+{
+	SECTION("Attribute unavailable") {
+		const auto e = MatcherException::from_rust_error(
+				rs_get_test_attr_unavail_error());
+		REQUIRE(e.type() == MatcherException::Type::ATTRIB_UNAVAIL);
+		REQUIRE(e.info() == "test_attribute");
+		REQUIRE(e.info2().empty());
```

Comment:
Might be useful to test the `what()` function as well.

Reply:
The message is localized, I try not to test on those. Can definitely check that the result is non-empty. Do you think that'd be enough?

Reply:
That's fine with me.

Reply:
Fixed.